packages/cli/src/commands/build/command.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command } from 'commander';
+import { buildPackage, Output } from '../../lib/builder';
+import { PackageRole, findRoleFromCommand, getRoleInfo } from '../../lib/role';
+
+const bundledRoles: PackageRole[] = ['app', 'backend'];
+
+const esmPlatforms = ['web', 'common'];
+const cjsPlatforms = ['node', 'common'];
```

Comment:
makes me wonder if it isn't actually more easily understood and debugged, albeit less pretty, to explicitly do

```ts
const roleDefaults = {
  app: {
    outputs: ['esm', 'types'],
  },
  backend: {
    outputs: ['cjs', 'types'],
  },
  // ... etc
} as const;
```

it'll be longer, but also a nice overview and less code in the end. And then command line switches can override parts of that structure etc. Could even consider putting different concerns in the same structure in a central place:

```ts
const roleDefaults = {
  app: {
    build: {
      outputs: ['esm', 'types'],
    },
    bundle: {
      configPaths: ['app-config.yaml'],
    }
  },
  // ... etc
} as const;
```

or even without that additional nesting 🤷

Reply:
Yep I think this could be moved into the `PackageRoleInfo` tbh, what outputs a package has may be a useful thing to know for other parts too. Could potentially switch out the `bundled: true` for an `outputs: ['bundle']` as well

Reply:
Part of the reason it's like this is because the outputs are an enum atm btw, and I'd really want to refactor that because it's just been annoying xP

Thinking it might be good to do in a followup though, since this already feels large enough